x/auth/mapper.go

```diff
@@ -142,30 +139,6 @@ func (am AccountMapper) GetNextAccountNumber(ctx sdk.Context) int64 {
 //----------------------------------------
 // misc.
 
-// Creates a new struct (or pointer to struct) from am.proto.
```

Comment:
was there some reason we didn't just do this initially ? 

Is it worth introducing a `type AccountConstructor func() Account` ?

Reply:
Seems this could be used in the unit tests as well

Reply:
`type ProtoAccount func() Account` looks better for me, I think word 'proto' fits with the context.